### DISCUSSION TITLE
Combobox SelectedItem returns null after selecting the same item twice

### DISCUSSION THREAD

**Discussion by malarikvilmos** [AvaloniaUI/Avalonia#9834] [Q&A] [answered]
**Describe the bug**
In a Combobox, while having its **Items** property bound to a collection and also having its **SelectedItem** bound to a string property, selecting the same item twice will update the **SelectedItem** property's value to **null**.

**To Reproduce**
The XAML:

```
<ComboBox Items="{Binding Drives}" SelectedItem="{Binding SelectedItem, Mode=TwoWay}" SelectedIndex="0">
    <ComboBox.ItemTemplate>
        <DataTemplate>
            <TextBlock>
                <TextBlock.Text>
                    <MultiBinding StringFormat="{}{0}{1}">
                        <Binding Path="Name" FallbackValue="C:\"/>
                        <Binding Path="VolumeLabel" StringFormat="  | {0}" FallbackValue=""/>
                    </MultiBinding>
                </TextBlock.Text>
            </TextBlock>
        </DataTemplate>
    </ComboBox.ItemTemplate>
</ComboBox>
```

The ViewModel code:

```
public static List<DriveInfo> Drives => new(DriveInfo.GetDrives());

private string selectedItem;
public string SelectedItem
{
    get => selectedItem;
    set
    {
        if (value != null)
            selectedItem = value;
        else 
            System.Console.WriteLine("It was null again.");
     }
}
```

**Expected behavior**
**value** in this case should be the item selected, not null.

**Desktop:**

- OS: Windows 11
- Version 0.10.18

**Comment by timunie** [comment, marked as answer]
I don't get your usecase. If Items is a list of `DriveInfo`, how should selected item be `string`? It should be `DriveInfo` as well I guess.

**Comment by malarikvilmos** [comment]
Well that does seem to work out better. Also gets rid of all the automatic string conversions.
